src/score_hv/region_utils.py

```diff
@@ -0,0 +1,126 @@
+"""
+Copyright 2024 NOAA
+All rights reserved.
+
+Collection of methods to work with user requested
+geographic regions.
+"""
+import sys,os
+import numpy as np
+import xarray as xr
+import math
+from datetime import datetime
+import pytest
+import pdb
+from netCDF4 import Dataset
+from pathlib import Path
+"""
+  The class GeoRegions has functions to allow the user to request specific
+  geographical regions.
+  """
+
+DEFAULT_MAX_LAT = 90.0
+DEFAULT_MIN_LAT = -90.0
+DEFAULT_EAST_LON = 0.0
+DEFAULT_WEST_LON = 360.0
+
+class GeoRegions:
+    def __init__(self,latitudes,longitudes):
+        """
+          Parameters:
+          - new_name: Name of the new region.
+          - new_min_lat: Minimum latitude of the new region.
+          - new_max_lat: Maximum latitude of the new region.
+          - new_east_lon: Eastern longitude of the new region.
+          - new_west_lon: Western longitude of the new region.
+        """
+        self.name=[]
+        self.min_lat=[]
+        self.max_lat=[]
+        self.east_lon=[]
+        self.west_lon=[]
+        self.latitudes=latitudes
+        self.longitudes=longitudes
+
+    def add_user_region(self, new_name, new_min_lat, new_max_lat, \
+                        new_east_lon, new_west_lon):
+        """
+          Calculate a new region based on specified parameters.
+
+          Parameters:
+          - new_name: Name of the new region.
+          - new_min_lat: Minimum latitude of the new region.
+          - new_max_lat: Maximum latitude of the new region.
+          - new_east_lon: Eastern longitude of the new region.
+          - new_west_lon: Western longitude of the new region.
+        """
+        if not isinstance(new_name, str):
+           msg = f'name must be a string - name {new_name}'
+           raise ValueError(msg)
+        
+        if new_min_lat > new_max_lat:  
+           msg = f'minimum latitude must be less than maximum latitude ' \
+                 f'min_lat: {new_min_lat}, max_lat: {new_max_lat}'  
+           raise ValueError(msg)
+         
+        if new_min_lat < -90. or new_min_lat > 90.:
+           msg = f'minimum latitude must be greater than -90. and less than 90. '\
```

Comment:
valid input lats could (should?) also include ints, not just floats, and the code should work with either

```suggestion
           msg = f'minimum latitude must be greater than or equal to -90 and less than or equal to 90 (degrees North)'\
```